### DISCUSSION TITLE
CAP theorem

### DISCUSSION THREAD

**Discussion by adambajguz** [ravendb/ravendb#19558] [General]
Hi RavenDB team,
recently I was trying to summarize the CAP theorem in terms of RavenDB, and I modified a widly known image from the Internet. **Am I thinking right?**
 
![CAP RavenDB](https://github.com/user-attachments/assets/cba874f5-224f-49a3-94c2-0e39e15232c8)

**Comment by karmeli87** [comment]
Hi :)

My understanding of the CAP theorem is that, in the event of a network partition in distributed systems, you can prioritizing either Availability or Consistency.

- CA systems are possible only when no network partition is involved, so I would say it is a Single node of RavenDB.
- AP is for prioritizing Availability, which is the master-master replication between the database nodes, which always accept writes and might result in conflicts and inconsistent data between the nodes. 
In the RavenDB context those are almost all database related operations, such as document CRUD (non cluster-wide transaction), querying, patching, etc.
- CP systems prefer Consistency and in RavenDB we rely on the RAFT algorithm to achieve this. So operations such as adding new nodes to the database, creating the databases, indexes, other settings and configuration has to be consistent. Also for document you can prefer consistency over availability by using the cluster-wide transactions.

**Reply by adambajguz** [reply]
Thanks. You're right CA should be single node, my mistake.

I'll close the discussion.

**Reply by adambajguz** [reply]
![CAP RavenDB](https://github.com/user-attachments/assets/703a5446-ef08-49ff-a347-121a3a5a6fbc)
